PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/StorageUploadWithoutLengthCheck.java

```diff
@@ -0,0 +1,125 @@
+    }
+
+    private static final List<String> METHODS_TO_CHECK = getMethodsToCheck();
+
+    private static List<String> getMethodsToCheck() {
+        try {
+            String configFileName = "META-INF/ruleConfigs.json";
+            InputStream inputStream = StorageUploadWithoutLengthCheck.class.getClassLoader().getResourceAsStream(configFileName);
+            if (inputStream == null) {
+                throw new FileNotFoundException("Configuration file not found");
+            }
+            try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
+                JSONObject jsonObject = new JSONObject(bufferedReader.lines().collect(Collectors.joining()));
+                JSONArray methods = jsonObject.getJSONObject("StorageUploadWithoutLengthCheck").getJSONArray("methodsToCheck");
+                return methods.toList().stream().map(Object::toString).collect(Collectors.toList());
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            return Collections.emptyList();
+        }
+    }
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new JavaRecursiveElementWalkingVisitor() {
+
+            @Override
+            public void visitMethodCallExpression(PsiMethodCallExpression expression) {
+                super.visitMethodCallExpression(expression);
+                String methodName = expression.getMethodExpression().getReferenceName();
+
+                if (METHODS_TO_CHECK.contains(methodName)) {
+                    boolean hasLengthArg = false;
+
+                    PsiExpressionList argumentList = expression.getArgumentList();
+                    PsiExpression[] arguments = argumentList.getExpressions();
+
+                    for (PsiExpression arg : arguments) {
+                        // Check if the argument is of type 'long'
+                        if (arg.getType().toString().equals("PsiType:long")) {
+                            hasLengthArg = true;
+                            break;
+                        }
+                        if (arg instanceof PsiMethodCallExpression) {
+                            PsiMethodCallExpression argMethodCall = (PsiMethodCallExpression) arg;
+                            PsiExpression qualifier = argMethodCall.getMethodExpression().getQualifierExpression();
+
+                            // Analysing arguments that are chained method calls
+                            hasLengthArg = checkMethodCallChain(argMethodCall);
+                        }
+                    }
+                    if (!hasLengthArg) {
+                     holder.registerProblem(expression, "Azure Storage upload API without length parameter detected");
+                    }
+                }
+            }
+        };
+    };
+
+    // Analysing arguments that are chained method calls
```

Comment:
We usually use "Documentation comments" for methods/documenting source code.
https://www.tutorialspoint.com/java/java_comments.htm
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html

Reply:
Also this comment doesn't help clarify what the method is doing exactly.

Reply:
added more comments & javadocs:)